test/integration/concrete/lockup/withdraw/withdraw.t.sol

```diff
@@ -40,108 +40,127 @@ abstract contract Withdraw_Integration_Concrete_Test is Integration_Test, Withdr
         external
         whenNotDelegateCalled
         givenNotNull
         givenStreamNotDepleted
-        whenCallerUnauthorized
+        whenToNonZeroAddress
+        whenWithdrawAmountNotZero
     {
-        // Make Eve the caller in this test.
-        changePrank({ msgSender: users.eve });
-
-        // Run the test.
-        uint128 withdrawAmount = defaults.WITHDRAW_AMOUNT();
+        uint128 withdrawableAmount = 0;
         vm.expectRevert(
-            abi.encodeWithSelector(Errors.SablierV2Lockup_Unauthorized.selector, defaultStreamId, users.eve)
+            abi.encodeWithSelector(
+                Errors.SablierV2Lockup_Overdraw.selector, defaultStreamId, MAX_UINT128, withdrawableAmount
+            )
         );
-        lockup.withdraw({ streamId: defaultStreamId, to: users.recipient, amount: withdrawAmount });
+        lockup.withdraw({ streamId: defaultStreamId, to: users.recipient, amount: MAX_UINT128 });
     }
 
-    function test_RevertWhen_FormerRecipient() external whenNotDelegateCalled givenNotNull givenStreamNotDepleted {
-        // Transfer the stream to Alice.
-        lockup.transferFrom(users.recipient, users.alice, defaultStreamId);
-
-        // Run the test.
-        uint128 withdrawAmount = defaults.WITHDRAW_AMOUNT();
-        vm.expectRevert(
-            abi.encodeWithSelector(Errors.SablierV2Lockup_Unauthorized.selector, defaultStreamId, users.recipient)
-        );
-        lockup.withdraw({ streamId: defaultStreamId, to: users.recipient, amount: withdrawAmount });
+    modifier whenNoOverdraw() {
+        _;
     }
 
-    function test_RevertWhen_ToZeroAddress()
+    function test_RevertWhen_CallerSender()
         external
         whenNotDelegateCalled
         givenNotNull
         givenStreamNotDepleted
-        whenCallerAuthorized
+        givenWithdrawalAddressNotRecipient
+        whenWithdrawAmountNotZero
+        whenNoOverdraw
     {
+        // Make the Sender the caller in this test.
+        changePrank({ msgSender: users.sender });
+
+        // Run the test.
         uint128 withdrawAmount = defaults.WITHDRAW_AMOUNT();
-        vm.expectRevert(Errors.SablierV2Lockup_WithdrawToZeroAddress.selector);
-        lockup.withdraw({ streamId: defaultStreamId, to: address(0), amount: withdrawAmount });
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                Errors.SablierV2Lockup_WithdrawalAddressNotRecipient.selector,
+                defaultStreamId,
+                users.sender,
+                users.sender
+            )
+        );
+        lockup.withdraw({ streamId: defaultStreamId, to: users.sender, amount: withdrawAmount });
     }
 
-    function test_RevertWhen_WithdrawAmountZero()
+    function test_RevertWhen_CallerUnknown()
         external
         whenNotDelegateCalled
         givenNotNull
         givenStreamNotDepleted
-        whenCallerAuthorized
-        whenToNonZeroAddress
+        givenWithdrawalAddressNotRecipient
```

Comment:
This has to be written below `whenNoOverdraw`.

Reply:
Agree.